kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/filter/EventLoopFilterThreadExecutor.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal.filter;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import io.netty.channel.EventLoop;
+
+import io.kroxylicious.proxy.filter.FilterThreadExecutor;
+
+public class EventLoopFilterThreadExecutor implements FilterThreadExecutor {
+
+    private final EventLoop eventLoop;
+
+    public EventLoopFilterThreadExecutor(EventLoop eventLoop) {
+        this.eventLoop = eventLoop;
+    }
+
+    @Override
+    public boolean isInFilterThread() {
+        return eventLoop.inEventLoop();
+    }
+
+    @Override
+    public ScheduledFuture<?> schedule(Runnable command, long delay, TimeUnit unit) {
+        return eventLoop.schedule(command, delay, unit);
+    }
+
+    @Override
+    public <V> ScheduledFuture<V> schedule(Callable<V> callable, long delay, TimeUnit unit) {
+        return eventLoop.schedule(callable, delay, unit);
+    }
+
+    @Override
+    public ScheduledFuture<?> scheduleAtFixedRate(Runnable command, long initialDelay, long period, TimeUnit unit) {
+        return eventLoop.scheduleAtFixedRate(command, initialDelay, period, unit);
+    }
+
+    @Override
+    public ScheduledFuture<?> scheduleWithFixedDelay(Runnable command, long initialDelay, long delay, TimeUnit unit) {
+        return eventLoop.scheduleWithFixedDelay(command, initialDelay, delay, unit);
+    }
+
+    @Override
+    public void shutdown() {
+        eventLoop.shutdown();
+    }
+
+    @Override
+    public List<Runnable> shutdownNow() {
+        return eventLoop.shutdownNow();
```

Comment:
Obviously the same logic as https://github.com/kroxylicious/kroxylicious/pull/965/files#r1484706382 applies here.